Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/preprocessor/LibraryInfo.java

```diff
@@ -224,11 +227,17 @@ public String resolveExpressionName(String identifier) {
     public void addFunctionDefinition(FunctionDefinitionInfo functionDefinition) {
         List<FunctionDefinitionInfo> infos = functionDefinitions.get(functionDefinition.getName());
         if (infos == null) {
-            infos = new ArrayList<FunctionDefinitionInfo>();
+            infos = new ArrayList<>();
             functionDefinitions.put(functionDefinition.getName(), infos);
         }
-        infos.add(functionDefinition);
-        addDefinition(functionDefinition);
+
+        // Ensure no duplicate FunctionDefinitionInfos (meaning the same signature)
+        if (! infos.contains(functionDefinition)) {
```

Comment:
issue: A duplicate should throw a compiler error. The implication here is the source CQL text contained overloads that collided. In a code editor you'd want to see the colliding functions highlighted with squiggles. The way that's surfaced in the compiler is with exceptions.